react/features/mobile/external-api/middleware.ts

```diff
@@ -186,18 +193,45 @@ externalAPIEnabled && MiddlewareRegistry.register(store => next => action => {
     }
 
     case ENDPOINT_MESSAGE_RECEIVED: {
-        const { participant, data } = action;
+        const { participant, data: json } = action;
+        const { text } = json.transcript[0];
+        const STABLE_TRANSCRIPTION_FACTOR = 0.85;
 
-        if (data?.name === ENDPOINT_TEXT_MESSAGE_NAME) {
+        if (json?.name === ENDPOINT_TEXT_MESSAGE_NAME) {
             sendEvent(
                 store,
                 ENDPOINT_TEXT_MESSAGE_RECEIVED,
                 /* data */ {
-                    message: data.text,
+                    message: json.text,
                     senderId: participant.getId()
                 });
         }
 
+        const { skipInterimTranscriptions }
+            = store.getState()['features/base/config'].testing ?? {};
+
+        if (json?.is_interim && skipInterimTranscriptions) {
+            break;
+        }
+
+        const txt: any = {};
+
+        if (!json?.is_interim) {
+            txt.final = text;
+        } else if (json?.stability > STABLE_TRANSCRIPTION_FACTOR) {
+            txt.stable = text;
+        } else {
+            txt.unstable = text;
+        }
+
+        sendEvent(
+            store,
+            TRANSCRIPTION_CHUNK_RECEIVED,
+            {
```

Comment:
@saghul I tried adding more props to this object, so we can log like we do it on web,  but there seems to be a RN/Strophe.js batched bridge message about 
" Malformed calls from JS: Field sizes are different "

Reply:
You mean you wanted to send multiple things in the object or as different arguments?

Reply:
Multiple things in the object, same structure like on web.

Reply:
Can you show the sample code that failed?

Reply:
{
      messageID: message_id,
      language,
      participant,
      ...txt
}

Reply:
Hum that's weird. Log the payload first, there might be some data type that cannot be serialized...